source/_posts/2020-02-18-wasm-ewasm-what-and-why.md

```diff
@@ -0,0 +1,120 @@
+
+*Metering injector is a transformation tool inserting metering code to an eWasm contract.
+
+**Toolchain Compatibility:** A LLVM front-end for Wasm is part of the MVP. This will allow developers to write contracts and reuse applications written in common languages such as C/C++, go and rust.
+
+**Portability**: Wasm is targeted to be deployed in all the major web browsers which will result in it being one of the most widely deployed VM architecture. Contracts compiled to eWasm will share compatibility with any standard Wasm environment. Which will make running a program either directly on Ethereum, on a cloud hosting environment, or on one's local machine - a frictionless process.
+
+**Optional And Flexible Metering:** Metering the VM adds overhead but is essential for running untrusted code. If code is trusted, then metering maybe optional. eWasm defines metering as an optional layer to accommodate for these usecases.
+
+
+## eWasm Performance
+
+This chart shows the Wasm-based EVM setting new performance records in a typical   sha1 benchmark experiment.  It is shown here against *Evmone* (a C++ implementation) and *Cita-vm* (a rust implementation).
+
+![Wasm EVM benchmarking](/assets/images/wasm-evm-benchmarks.png)  
+
+I took this benchmark directly from the eWasm GitHub, I didn't run the tests myself. You can find the full details, alongside more benchmark testing [here.](https://github.com/eWasm/benchmarking#sha1)
+
+
+
+## Conclusion
+
+Since its launch a couple of years ago; *Wasm* has grown from an impressive concept into an even-more-impressive piece of technology.  Alongside which a flourishing community has grown.  
+
+Although, personally, I have my reservations in agreeing with the statement that *Wasm* is the "most significant new technology to come to the web platform in a decade"...   It definitely **is** impressive, and I can certainly see how & why it has made a good impact, and does have a lot of devs working on relative tooling and implementations.  This was an interesting (albeit old) article I just stumbled across this morning:  [Life: A secure, blazing-fast, cross-platform WebAssembly VM in Go.](https://medium.com/perlin-network/life-a-secure-blazing-fast-cross-platform-webassembly-vm-in-go-ea3b31fa6e09)
+
+Between Wasm & eWasm, I'm excited to see the performance boosts we can achieve as they grow, and as our own DApp development grows.  I'd love to find an excuse to build out a big piece of technology with eWasm, but this *would* be a big project that I just don't have time for right now!
+
+Thanks again for reading my series on DApp frontend.  The next (and unfortunately) final article in the series will be released next week, and is sure to be a **real humdinger!**  We'll be building a realtime crypto/blockchain tracker app using Elixir & [Phoenix's](https://www.phoenixframework.org/) newly popular [LiveView!](https://github.com/phoenixframework/phoenix_live_view)
+
```

Comment:
@emizzle and I gave a talk about Ewasm and Ewasm inside Embark at last devcon. Maybe it's worth mentioning this?

Could drive people to trying out Embark (even though it was experimental). Here's the recording: https://www.youtube.com/watch?v=t2LgFXxcFtc